artifactory/services/utils/artifactoryutils.go

```diff
@@ -277,6 +320,73 @@ func filterAqlSearchResultsByBuild(specFile *ArtifactoryCommonParams, itemsToFil
 	return filterBuildAqlSearchResults(&itemsToFilter, &buildArtifactsSha1, buildName, buildNumber), err
 }
 
+// cr - File of sorted result item without properties
+// crWithProps - File of result item with properties
+// Load all properties to the sorted result items. Save the new result items to a file.
+// return a content reader which point to the result file.
+func loadMissingProperties(cr *content.ContentReader, crWithProps *content.ContentReader) (*content.ContentReader, error) {
+	// Key ->.Repo + .Path + Name + Actual_Sha1, value -> ResultItem
+	// Contain limited amount of items from a file, to not overflow memory.
+	buffer := make(map[string]*ResultItem)
+	var err error
+	// Create New file to write result output
+	resultFile, err := content.NewContentWriter("results", true, false)
+	if err != nil {
+		return nil, err
+	}
+	bufferCounter := 0
+	for resultItem := new(ResultItem); cr.NextRecord(resultItem) == nil; resultItem = new(ResultItem) {
+		// save the item in a buffer.
+		if bufferCounter < 50000 {
+			buffer[getResultItemKey(*resultItem)] = resultItem
+		} else {
+			// Buffer was full, write all data to a file.
+			err = UpdatePropsAndSaveToFile(crWithProps, resultFile, buffer)
+			if err != nil {
+				return nil, err
+			}
+			// Init buffer.
+			bufferCounter = 1
+			buffer = make(map[string]*ResultItem)
+			buffer[getResultItemKey(*resultItem)] = resultItem
+		}
+	}
+	if err = cr.GetError(); err != nil {
+		return nil, err
+	}
+	if err := UpdatePropsAndSaveToFile(crWithProps, resultFile, buffer); err != nil {
+		return nil, err
+	}
+	resultFile.Close()
+	// TODO: Remove this
+	// cr.Close()
+	// crWithProps.Close()
+	return content.NewContentReader(resultFile.GetFilePath(), "results"), nil
+}
+
+// buffer - hold limit amount of items (sorted)
+// crWithProps - file which contain all the results with proprties
+// cw - file to write sorted result item with properties
+func UpdatePropsAndSaveToFile(crWithProps *content.ContentReader, cw *content.ContentWriter, buffer map[string]*ResultItem) error {
```

Comment:
Looks like this function can be made private and its name can be shortened to *updateProps*.